### ISSUE TITLE
Lot of false negative

### ISSUE THREAD

**Issue by Willyfab** [rebryk/profanity-brute-force#13]
Hello,

I was testing your tool with already hacked addresses to understand how it is working (Mine for example) and some of them can't be cracked by the tool. Any idea why?

**Comment by Web3Paradise** [comment]
I've noticed the same but I have no idea why
Some addresses already cracked can't be cracked by the tool. Eight character work sometimes with 5000 steps, so consider checking either character adress only and increase the number of steps if it doesn't work. Anything under eight character or above digit 3 doesn't work for me (eg: 0x33333333, 0x77777777)

If you have a powerful GPU like RX 4090 or Apple M2 Ultra, you can try running with 500k - 2 million steps, maybe it'll work

Note: there are hundreds of profanity adress generator and most of them are not vulnerable so can't be cracked by this tool.

**Comment by babytether111** [comment]
Care to share an address you are getting a false negative for? I might be able to help.

**Comment by Web3Paradise** [comment]
> Hello,
> 
> I was testing your tool with already hacked addresses to understand how it is working (Mine for example) and some of them can't be cracked by the tool. Any idea why?

@Willyfab

You need to increase the steps. I've managed to crack some already hacked adresses, few doesn't work on low steps, other work on higher steps.


Try with at least 50k - 100k steps if you have high end GPU (RTX 4090 - Apple m2 max 38 gpu) or up to 500k steps on less powerful GPU.

**Comment by Willyfab** [comment]
> Care to share an address you are getting a false negative for? I might be able to help.

Don't remember which one but one of these two addresses can't be cracked


0x0Babe55554119E8b345e03b95cBebD84008feB05
0x0Babe3333bb2904DC3cdC16b80b64DC3Ec5Ac4d3

**Comment by babytether111** [comment]
Both of these addresses have been cracked by multiple exploiters, and I was able to crack both of them. 

I'd try renting a GPU if you are having an issue.

**Comment by Willyfab** [comment]
> Both of these addresses have been cracked by multiple exploiters, and I was able to crack both of them. 
> 
> 
> 
> I'd try renting a GPU if you are having an issue. 

With how many steps?

**Comment by Web3Paradise** [comment]
> Both of these addresses have been cracked by multiple exploiters, and I was able to crack both of them.
> 
> I'd try renting a GPU if you are having an issue.

If you were able to crack them, what are the private keys for both addresses? What's your GPU? How many steps?

**Comment by b006** [comment]
Never post private keys. 

Are these your addresses?
Both of these are at least 10 scores and should take on average 200k steps. it seems the 3333 address was lucky and could've been found very early for a 10 score.  As for 5555 its possible it was found way beyond 200k steps if it was a actually an 11+ score.

This tool was probably the 4th created other than the 0days so you have to factor in errors in the code too.

**Comment by Willyfab** [comment]
> Never post private keys. 
> 
> 
> 
> Are these your addresses?
> 
> Both of these are at least 10 scores and should take on average 200k steps. it seems the 3333 address was lucky and could've been found very early for a 10 score.  As for 5555 its possible it was found way beyond 200k steps if it was a actually an 11+ score.
> 
> 
> 
> This tool was probably the 4th created other than the 0days so you have to factor in errors in the code too.
> 
> 
> 
> 
> 
> 

Thanks for your answer! How do you determine steps and score?

**Comment by b006** [comment]
score is how many matches in the address
those addresses are at least a 10 because 0Babe3333.....3 is 10 matches to find
steps or rounds is how deep profanity had to go to find a match
a 10 score is is 16^10 in complexity meaning it'll on average find it within 1,099,511,627,776 tries
if profanity can do 500,000,000 tries a second that means it should on average find a match in 2200 seconds but there are obvious outliers

finding steps is just an educated guess you get with benchmarking

**Comment by babytether111** [comment]
@b006 I would also be very curious how you calculate steps based on number on scores

**Comment by Web3Paradise** [comment]
There may be some error with this tool. I've spent 8 hours trying to brute force a 9 char adress already cracked multiple times by others, and this tool failed to generate the key, even with 1 million steps. Perhaps a search issue? Some work, some doesn't, even if it's already cracked by others. I wonder which tool other people used.

**Comment by babytether111** [comment]
@Web3Paradise have been running into similar issues even with 1 million steps, primarily around known vulnerable contracts...

**Comment by b006** [comment]
the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision
a 10 score would be
16^10 / 500m / ~0.01s = 220000

most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result

the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps

you can practise on this guys address/contract targets:
https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442

since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.

**Comment by Web3Paradise** [comment]
> the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision a 10 score would be 16^10 / 500m / ~0.01s = 220000
> 
> most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result
> 
> the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps
> 
> you can practise on this guys address/contract targets: https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442
> 
> since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.

@b006 
I've been practicing on this guy target adresses for a while, it help me to determine how many steps is needed for X char/address. Most of his target can be cracked with 150k steps on RTX 4090 (1gh/s - 1350 mh/s).
He's using a smart contract to automatically wipe any funds deposited in these addresses to his adress. He doesn't care if an adress is already hacked. 100% of the adresses his is hacking are doomed forever.

**Comment by babytether111** [comment]
> > the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision a 10 score would be 16^10 / 500m / ~0.01s = 220000
> > most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result
> > the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps
> > you can practise on this guys address/contract targets: https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442
> > since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.
> 
> @b006 I've been practicing on this guy target adresses for a while, it help me to determine how many steps is needed for X char/address. Most of his target can be cracked with 150k steps on RTX 4090 (1gh/s - 1350 mh/s). He's using a smart contract to automatically wipe any funds deposited in these addresses to his adress. He doesn't care if an adress is already hacked. 100% of the adresses his is hacking are doomed forever.

hey web3paradise, I'd love to chat with you about a few issues I've been facing and compare notes, if you are open please drop a telegram

**Comment by Web3Paradise** [comment]
> > > the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision a 10 score would be 16^10 / 500m / ~0.01s = 220000
> > > most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result
> > > the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps
> > > you can practise on this guys address/contract targets: https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442
> > > since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.
> > 
> > 
> > @b006 I've been practicing on this guy target adresses for a while, it help me to determine how many steps is needed for X char/address. Most of his target can be cracked with 150k steps on RTX 4090 (1gh/s - 1350 mh/s). He's using a smart contract to automatically wipe any funds deposited in these addresses to his adress. He doesn't care if an adress is already hacked. 100% of the adresses his is hacking are doomed forever.
> 
> hey web3paradise, I'd love to chat with you about a few issues I've been facing and compare notes, if you are open please drop a telegram

You can leave your telegram username, I'll Dm you

**Comment by babytether111** [comment]
@Web3Paradise i'm @anonresearcher on telegram

**Comment by Willyfab** [comment]
> > the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision a 10 score would be 16^10 / 500m / ~0.01s = 220000
> 
> > 
> 
> > most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result
> 
> > 
> 
> > the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps
> 
> > 
> 
> > you can practise on this guys address/contract targets: https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442
> 
> > 
> 
> > since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.
> 
> 
> 
> @b006 
> 
> I've been practicing on this guy target adresses for a while, it help me to determine how many steps is needed for X char/address. Most of his target can be cracked with 150k steps on RTX 4090 (1gh/s - 1350 mh/s).
> 
> He's using a smart contract to automatically wipe any funds deposited in these addresses to his adress. He doesn't care if an adress is already hacked. 100% of the adresses his is hacking are doomed forever. 

I can crack 90% of this guy targets with 20000 steps using a M2 Max GPU.

I will now increase the number of steps for the missing ones.

I read someone said 99.5% of profanity wallet have already been hacked. Do you think people already scanned millions of wallets?

**Comment by babytether111** [comment]
Yes, its trivial to search for vanity addresses using dune, and almost certainly most of the vulnerable addresses have been exploited. 

All of the recent exploits I've noticed have been people deploying contracts on other chains where erc20 and ETH/BNB has been "stuck" in addresses, and then withdrawing that ETH. Last big one (4k USD) was ENS 26 days ago

**Comment by b006** [comment]
that case seems unique because the tokens were recently sent to that address and it was worth $0 when the first couple waves of bots checked. a lucky situation.

@Willyfab this exploit has been public for more than a year now. 0day: hundreds of millions $ were out there. After this tool: maybe millions were still left. a year later? maybe $10k scattered in small amounts other than when people accidentally send funds to a compromised address. gpu farms of A100s have already scanned every address with large $ in it. youre late to the party

so going back to the issue. unless someone wants to prove the number of steps that 0x0babe5555 address takes I think this issue can be closed.
tool works as intended with no demonstrable false negatives

**Comment by Web3Paradise** [comment]
> > > the score possibilities (permutations with repetitions set) divided by your hashrate divided by your benchmarked seconds per step will return an estimated # of steps to find a collision a 10 score would be 16^10 / 500m / ~0.01s = 220000
> > 
> > 
> > > 
> > 
> > 
> > > most addresses were skewed to a lower number of steps because people would just restart profanity after a certain amount of time instead of waiting for a result
> > 
> > 
> > > 
> > 
> > 
> > > the 5555 is obviously an abnormality so good luck to anyone that wants to waste time finding the right # of steps
> > 
> > 
> > > 
> > 
> > 
> > > you can practise on this guys address/contract targets: https://etherscan.io/address/0xfe573bc39202180cf6653b99da88c68c48c57442
> > 
> > 
> > > 
> > 
> > 
> > > since 99.99% of profanity addresses have been compromised by now I dont really care enough to test more. try generating a hard address on profanity 1.6 then use this tool to confirm your suspicions.
> > 
> > 
> > @b006
> > I've been practicing on this guy target adresses for a while, it help me to determine how many steps is needed for X char/address. Most of his target can be cracked with 150k steps on RTX 4090 (1gh/s - 1350 mh/s).
> > He's using a smart contract to automatically wipe any funds deposited in these addresses to his adress. He doesn't care if an adress is already hacked. 100% of the adresses his is hacking are doomed forever.
> 
> I can crack 90% of this guy targets with 20000 steps using a M2 Max GPU.
> 
> I will now increase the number of steps for the missing ones.
> 
> I read someone said 99.5% of profanity wallet have already been hacked. Do you think people already scanned millions of wallets?

Yes, 99% of the wallets have already been hacked, it will be really hard to find a wallet that's not empty. Most of the hack happened a year ago. The ones remaining are not vulnerable because they were created using other vanity address generator tools. Good luck finding a wallet that's not already cracked.
Also, never try to deposit in a profanity wallet from now on, your funds will disappear automatically. Everyone is using a smart contract sweeper bot on wallet already cracked. I've tried to deposit $1 in a wallet already hacked and it was gone before landing lol

**Comment by Web3Paradise** [comment]
> that case seems unique because the tokens were recently sent to that address and it was worth $0 when the first couple waves of bots checked. a lucky situation.
> 
> @Willyfab this exploit has been public for more than a year now. 0day: hundreds of millions $ were out there. After this tool: maybe millions were still left. a year later? maybe $10k scattered in small amounts other than when people accidentally send funds to a compromised address. gpu farms of A100s have already scanned every address with large $ in it. youre late to the party
> 
> so going back to the issue. unless someone wants to prove the number of steps that 0x0babe5555 address takes I think this issue can be closed. tool works as intended with no demonstrable false negatives

Yes, this tool work as intended...
I can probably brute force 0x0babe5555 with 400k+ steps but don't want to waste my time and my computer resources..
My RTX 4090 take ~3h with 350k steps (I did a comparison between pub key generation times and the number of steps required to do the same, it was 350k steps from batch 1 to 8) to precompute all public Keys

**Comment by Web3Paradise** [comment]
I've challenged myself to brute force 10+ char, and it worked lol. You need to consider running up to 2,500,000 steps and waste almost 20h with 4090...
1h30/batch , if you're very lucky, you'll find the key in batch 1 or 2 and won't waste (?) 20h
 I guess 12 char should take up to 8 million steps, 4 hours to check 1 batch. Unless it's worth it, don't waste your time on low steps trying to brute force 10+ char 
Would be great if we have multiple GPU support for this tool.

**Comment by Willyfab** [comment]
Thanks for your reply!
Last question. What is the MH/S with a RTX 4090?

**Comment by Web3Paradise** [comment]
> Thanks for your reply! Last question. What is the MH/S with a RTX 4090?

1400 mh/s (1,39 gh/s)

**Comment by ytrezq** [comment]
@Web3Paradise : because of the Makefile, when you build on Mac, is it g++ or clang++ ?
